peachjam/views/documents.py

```diff
@@ -9,6 +11,55 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {"code": "aa", "domain": "agp.africanlii.org"},
+        "ghalii": {"code": "gh", "domain": "new.ghalii.org"},
+        "lawlibrary": {"code": "za", "domain": "lawlibrary.org.za"},
+        "lesotholii": {"code": "ls", "domain": "new.lesotholii.org"},
+        "malawilii": {"code": "mw", "domain": "malawilii.org"},
+        "namiblii": {"code": "na", "domain": "namiblii.org"},
+        "senlii": {"code": "sn", "domain": "senlii.org"},
+        "seylii": {"code": "sc", "domain": "new.seylii.org"},
+        "sierralii": {"code": "sl", "domain": "new.sierralii.org"},
+        "tanzlii": {"code": "tz", "domain": "new.tanzlii.org"},
+        "tcilii": {"code": "tc", "domain": "new.tcilii.org"},
+        "ulii": {"code": "ug", "domain": "new.ulii.org"},
+        "zambialii": {"code": "zm", "domain": "new.zambialii.org"},
+        "zanzibarlii": {"code": "tz-znz", "domain": "new.zanzibarlii.org"},
+        "zimlii": {"code": "zw", "domain": "zimlii.org"},
+    }
+
+    def __init__(self):
+        self.current_authority = self.RESOLVER_MAPPINGS[
+            settings.PEACHJAM["APP_NAME"].lower()
+        ]
+
+    def get_url_for_frbr_uri(self, frbr_uri):
+        try:
+            parsed_uri = FrbrUri.parse(frbr_uri)
```

Comment:
I think the caller should do the parsing and provide an FrbrUri object. This object doesn't have enough context to know what to do if the FRBR URI isn't valid, whereas the caller has a better idea (eg. raise a 404 or throw an exception).

Reply:
added